### ISSUE TITLE
[2019.1.0a12] missing reference to UnitySpatialTracking

### ISSUE THREAD

**Issue by iBicha** [Unity-Technologies/arfoundation-samples#79]
Empty new project with 2019.1.0a12, then adding AR Foundation from package manager:

```
Library/PackageCache/com.unity.xr.arfoundation@1.0.0-preview.22/Runtime/AR/ARSessionOrigin.cs(3,19): error CS0234: The type or namespace name 'SpatialTracking' does not exist in the namespace 'UnityEngine' (are you missing an assembly reference?)
```

I'm guessing this because 2019 is missing the `UnitySpatialTracking` extension (which is found in 2018.3, where this error doesn't occur)

**Comment by tdmowrer** [comment]
This is because the `TrackedPoseDriver` has been moved to a package in 2019.1, and we will need a separate package version to support it (the same package cannot be used for both 2018.3 and 2019.1, and there is no way to specify a maximum Unity version).

If you want to try the bleeding edge now, you'll need to add this package to your project's `manifest.json`:
```json
"com.unity.xr.legacyinputhelpers": "1.0.0"
```

and also add a reference to `UnityEngine.SpatialTracking` to ARFoundation's asmdef.

**Comment by malDuffin** [comment]
Will this be fixed for the next AR Foundation update?

**Comment by ZeKJ** [comment]
Any news? :)

**Comment by immemorable-moniker** [comment]
still present with 2019.1.0b7

**Comment by jackpr-unity** [comment]
@tdmowrer packages through package manager are read-only by default and changes to them are "fixed" at next editor load.  This project probably has to be updated to 19.1 (maybe leave behind a 18.3 branch for legacy support?)

**Comment by jackpr-unity** [comment]
FYI - arfoundation package version 1.1.0-preview.6 works with 2019.1.0b8.

**Comment by BrahimHadriche** [comment]
I think this is solved now. Please reopen if this still exists.

**Comment by iBicha** [comment]
I think this is fixed now. Please reopen if it's not

**Comment by julia1981** [comment]
This issue exists in 2019.1.0f2. How to fix this? Thanks

**Comment by tdmowrer** [comment]
@julia1981 Make sure you are on AR Foundation 1.1 not 1.0. Only 1.1 works with 2019.1.

**Comment by julia1981** [comment]
Thanks, did a update and it works

**Comment by neogeek** [comment]
Can confirm that it's fixed in `1.1.0-preview.6`, but the error is back in `1.5.0-preview.2`. I tried installing `com.unity.xr.legacyinputhelpers`, but was unclear as to how to properly modify the assembly file to link them.

**Comment by tdmowrer** [comment]
Use ARFoundation 1.5 for Unity 2018.3/4
Use ARFoundation 2.1 for Unity 2019.1/2

The only difference between ARFoundation 1.5 and 2.1 are the versions of Unity with which they are compatible. Unfortunately, there is no way for us to specify a _max_ version of Unity.

**Comment by neogeek** [comment]
Whoops, didn't even think of that. Thanks for the heads up!

**Comment by gblikas** [comment]
@tdmowrer If you create a unitypackage in Unity 2019.2.18f1 and load it in another project using Unity 2019.2.18f1, this still seems to be a problem; without adding `"com.unity.xr.legacyinputhelpers": "2.0.2"` to your manifest file, the AR Session Origin object is broken, as others have mentioned. 

Also, the script reference missing shows up as a warning, not an error-- I believe this should be an error as a TrackedPose driver not working in AR is a pretty fundamental mishap. The fact that nothing in any of the PackageManager descriptions says what you mention about 1.5 vs 2.1 is an oversight at the least. Aggressively, the PackageManager should prompt you to install `legacyinputhelpers`.

**Comment by tdmowrer** [comment]
We no longer depend on the tracked pose driver. You can still use it if you want, but ARFoundation has its own ARPoseDriver which provides similar functionality designed specifically to eliminate the dependency on legacy input helpers.

If you replace your tracked pose driver with an AR pose driver, that should fix the problem.

**Comment by Yo030** [comment]
There is a way around this error in Unity 2019.1.7f, all you have to do is to install AR Foundation, remove it and reinstall AR Foundation, for some reason this worked for me.

**Comment by allenWTF** [comment]
I have this error on Unity 2018.4.22f1 when trying to get WebGL build. I understand  that it might be because there is no AR for WebGL and i would be happy if I could just turn this package off or something but i don't get it how.
I've already tried to reinstall package, tried to install legacyinput handlers. Basically I tried all advices that can be related to this error (well, except for trying to built it through Unity 2019).
Any help will be appreciated! Thanks!
